### ISSUE TITLE
Add support for structured logging

### ISSUE THREAD

**Issue by vmartos** [BrighterCommand/Brighter#1254]
Hello,

We are using Brighter for the message bus in our application, we have centralized logging using elasticsearch + kibana. We saw that in certain fields there are type conflicts (long and string with the same field), all of them are related with Paramore.Brighter.* logs.

![image](https://user-images.githubusercontent.com/10826413/93856255-074f2c80-fcb9-11ea-8893-5a99d6a1f373.png)

My feature request is to use structured logging with the defacto standard Microsoft.Extensions.Logging.Abstractions, instead of LibLog and change the logs in a structured way (don't use {0}, {1}, etc logs), so there should not be problems with the same field having different types.

I already see there are intentions of removing LibLog: https://github.com/BrighterCommand/Brighter/issues/242

One example of structured logging could be:

instead of:
`_logger.Value.WarnFormat("RMQMessagingGateway: Connection to {0} blocked. Reason: {1}", 
                Connection.AmpqUri.GetSanitizedUri(), args.Reason);`
use:
`logger.LogInformation ("RMQMessagingGateway: Connection to {uri} blocked. Reason: {reason}", Connection.AmpqUri.GetSanitizedUri(), args.Reason);`

so basically is removing the {0}, {1}, etc in logs and substitue with fields, and try to use the same fields for the same data type.

If structured logging is not feasible, because some users do not use structured logging, maybe string interpolation could be a better option:

`_logger.Value.Warn($"RMQMessagingGateway: Connection to {Connection.AmpqUri.GetSanitizedUri()} blocked. Reason: {args.Reason}");`

I hope everything is clear, if you tackle https://github.com/BrighterCommand/Brighter/issues/242 please take in mind to also tackle this to have structured logging.

Thanks for this great library!

**Comment by iancooper** [comment]
We already have a discussion of the need to remove LibLog, but this is valuable input. We should continue the discussion under #

**Comment by iancooper** [comment]
Hey @holytshirt this is worth thinking about when we tackle replacing LibLog in #242

**Comment by preardon** [comment]
@iancooper @holytshirt I will have a PR in for this soon, one thing I would like to discuss are the Fields

At the minute I have used the following

- Id
- CommandType
- EventType
- HandlerType
- RequestType
- Type
- HandlerCount
- HandlerName
- NextHandler
- HandlerTiming
- Topic
- ChannelName
- Request
- OutstandingMessageCount
- SecondsSinceLastCheck
- MessageCheckTime
- InstanceHashCode
- Time

I wonder if I should Merge the 5 Types into Type also I'm not the biggest fan of Time as Loggers usually include these, Let me know what you think, I'm aiming to have the PR in very soon

**Comment by preardon** [comment]
@holytshirt I see You've already done Most of the ASB project, It it worth Standardising the Logging Fields? i.e. you've used Topic and TopicName for the same thing as well as using SubscriptionName (The ASB name for that component) where I've been using ChannelName (The Brighter Name) regardless?

**Comment by preardon** [comment]
PR #1499 in for this

**Comment by iancooper** [comment]
@preardon Can we list outstanding actions here please? Thanks, you are a star for tackling this one.

**Comment by preardon** [comment]
@iancooper I believe the only thing Left was for us to Review how the Keys I used and if we are Happy i.e. Having CommandType, HandlerType and RequestType

**Comment by iancooper** [comment]
@preardon Let me review. It occurs that channel name and routing key are different for us. Channel name is what we call your subscription i.e. queue name on some platforms, but routing key is what you subscribe to, i.e. topic on some platforms

**Comment by iancooper** [comment]
@preardon Going in, but it might take some time :-D

**Comment by iancooper** [comment]
So I went through all of these and made some comments for you to all review @holytshirt @preardon 
[brighter-logs.xlsx](https://github.com/BrighterCommand/Brighter/files/6426013/brighter-logs.xlsx)
